tests/components/slack/test_notify.py

```diff
@@ -93,3 +94,18 @@ async def test_message_icon_url_overrides_default() -> None:
     mock_fn.assert_called_once()
     _, kwargs = mock_fn.call_args
     assert kwargs["icon_url"] == expected_icon
+
+
+async def test_message_as_reply() -> None:
+    """Tests that a message pointer will be passed to Slack if specified."""
+    mock_client = Mock()
+    mock_client.chat_postMessage = AsyncMock()
+    service = SlackNotificationService(None, mock_client, CONF_DATA)
```

Comment:
We prefer setting up the integration and then calling the service and then assert that the library mock was called appropriately.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations